agimus_controller/tests/test_ocp_croco_base.py

```diff
@@ -0,0 +1,259 @@
+            terminal_cost_model.addCost("gripperPose", goal_tracking_cost, 50)
+
+            # Create Differential Action Model (DAM), i.e. continuous dynamics and cost functions
+            terminal_DAM = crocoddyl.DifferentialActionModelFreeFwdDynamics(
+                self._state,
+                self._actuation,
+                terminal_cost_model,
+            )
+
+            terminal_model = crocoddyl.IntegratedActionModelEuler(terminal_DAM, 0.0)
+            terminal_model.differential.armature = self._robot_models.armature
+            return terminal_model
+
+        def set_reference_horizon(self, horizon_size):
+            ### Not implemented in this OCP example.
+            return None
+
+        def update_crocoddyl_problem(self, x0, trajectory_points_list):
+            ### Not implemented in this OCP example.
+            return None
+
+    @classmethod
+    def setUpClass(self):
+        # Mock the RobotModelFactory and OCPParamsCrocoBase
+        robot = robex.load("panda")
+        urdf_path = Path(robot.urdf)
+        srdf_path = Path(robot.urdf.replace("urdf", "srdf"))
+        urdf_meshes_dir = urdf_path.parent.parent.parent.parent.parent
+        free_flyer = False
+        locked_joint_names = ["panda_finger_joint1", "panda_finger_joint2"]
+        reduced_nq = robot.model.nq - len(locked_joint_names)
+        full_q0 = np.zeros(robot.model.nq)
+        q0 = np.zeros(reduced_nq)
+        armature = np.full(reduced_nq, 0.1)
+        # Store shared initial parameters
+        self.params = RobotModelParameters(
+            q0=q0,
+            full_q0=full_q0,
+            free_flyer=free_flyer,
+            locked_joint_names=locked_joint_names,
+            urdf_path=urdf_path,
+            srdf_path=srdf_path,
+            urdf_meshes_dir=urdf_meshes_dir,
+            collision_as_capsule=True,
+            self_collision=True,
+            armature=armature,
+        )
+
+        self.robot_models = RobotModels(self.params)
+        self.robot_model = self.robot_models.robot_model
+        self.collision_model = self.robot_models.collision_model
+
+        # Set mock parameters
+        self.ocp_params = OCPParamsBaseCroco(
+            dt=0.1, horizon_size=10, solver_iters=100, callbacks=True
+        )
+        self.state_reg = np.concatenate(
+            (pin.neutral(self.robot_model), np.zeros(self.robot_model.nv))
+        )
+        self.state_warmstart = [np.zeros(self.robot_model.nq + self.robot_model.nv)] * (
+            self.ocp_params.horizon_size - 1
+        )  # The first state is the current state
+        self.control_warmstart = [np.zeros(self.robot_model.nq)] * (
+            self.ocp_params.horizon_size - 1
+        )
+        # Create a concrete implementation of OCPBaseCroco
+        self.ocp = self.TestOCPCroco(self.robot_models, self.ocp_params)
+        self.ocp.solve(self.state_reg, self.state_warmstart, self.control_warmstart)
+        # self.save_results()
+
+    @classmethod
```

Comment:
that is weird. A class method is supposed to be like a static class:
```suggestion
    @classmethod
    def save_results(cls, obj):
            results = np.array(
                [
                    obj.ocp.ocp_results.states.tolist(),
                    obj.ocp.ocp_results.ricatti_gains.tolist(),
                    obj.ocp.ocp_results.feed_forward_terms.tolist(),
                ],
                dtype=object,  # Ensure the array is dtype=object before saving
            )
            np.save(
                "ressources/simple_ocp_croco_results.npy",
                results,
            )
```